src/riak_cs_block_server.erl

```diff
@@ -240,16 +240,17 @@ do_get_block(ReplyPid, Bucket, Key, ClusterID, UseProxyGet, ProxyActive,
              UUID, BlockNumber, RcPid) ->
     MaxRetries = riak_cs_config:get_env(riak_cs, block_get_max_retries, 5),
     do_get_block(ReplyPid, Bucket, Key, ClusterID, UseProxyGet, ProxyActive,
-                 UUID, BlockNumber, RcPid, MaxRetries, 0).
+                 UUID, BlockNumber, RcPid, MaxRetries, []).
 
 do_get_block(ReplyPid, _Bucket, _Key, _ClusterID, _UseProxyGet, _ProxyActive,
-             UUID, BlockNumber, _RcPid, MaxRetries, NumRetries)
-  when is_atom(NumRetries) orelse NumRetries > MaxRetries ->
-    Sorry = {error, notfound},
+             UUID, BlockNumber, _RcPid, MaxRetries, ErrorReasons)
+  when is_list(ErrorReasons) andalso length(ErrorReasons) > MaxRetries ->
+    Sorry = {error, hd(ErrorReasons)},
```

Comment:
I wonder returning whole reasons or just head of this, which is better  ... `riak_cs_get_fsm` handles this Reason just for logging. So maybe this should be rather a list of errors than the head of errors.